### ISSUE TITLE
feat: finish depth interval related UI work on site soil tab

### ISSUE THREAD

**Issue by shrouxm** [techmatters/terraso-mobile-client#502]
```[tasklist]
### remaining work after #459
- [x] Finish styling the edit interval modal
- [x] Add proper error messages for the interval modal validation
- [x] Correctly display & disable the data input toggle for data inputs that are required at the project level in the site-level depth interval edit modal
- [x] Add a delete button to the edit interval modal for site-level depth intervals with associated modal warning
- [x] Wire up the "apply to all intervals" checkbox to actually do the thing
- [x] Add modal warning when changing a depth interval with existing soil data
- [x] There are probably a few other minor style issues to address besides the edit interval modal
- [x] Make radio switches save asynchronously to correct for lag
- [x] Restrict options based on preset  selected
- [x] Restrict options based on user role
```

```[tasklist]
### remaining work after feb 28
- [x] apply to all button doesn't work
- [x] delete button shouldn't exist for preset or project intervals
- [x] disabled forms are still editable / what to do about disabled forms
- [x] styling for required data input switches
- [x] label can show up for preset depth interval
- [x] add modal warning when deleting custom depths
- [x] remove cm from depth interval table
- [x] add modal warning when changing site depth preset
- [x] Check whether new sites have depth dependent data inputs enabled automatically
```

**Comment by david-code** [comment]
@ltseng For your visibility: I'm almost done with this screen, but there's a few final issues that are causing me some problems. I think it makes more sense to finish #381 , so we can see how it will work when the user selects presets at the project level. There's still some backend work that needs to be done as part of the same issue (#381) that will make finishing this issue more straightforward.

**Comment by ltseng** [comment]
Thanks for the update! This makes sense. Let me know if there's anything I can do to help shed clarity or unblock you otherwise.

**Comment by david-code** [comment]
@shrouxm Trying to think through project/site depth intervals. Writing this up so we can discuss next week. (Remember to also discuss deletion of site dependent data!)

A few points:

- Project depth intervals are not saved to the backend for the LandPKS or NRCS preset selections. 
    - These presets define a fixed set of depth interval bounds which can be generated on the frontend.
    - There is no need to save the required inputs to each interval. These are saved to the project soil settings.
- A given site intervals are made up of the list of project level intervals, as well as any user defined custom intervals
    - LandPKS and NRCS define intervals for the whole possible range, so that is not really an issue for them.
    - If there's a custom preset at the project level, it might not cover the whole range of possible depth input bounds. In that case a project manager or contributor can add site level intervals
    - If the site does not belong to a project, or the project has chosen the "None" preset, the only intervals will be those
       - In this case, a site owner / project manager or contributor can also select a site level preset for the site, either LandPKS, NRCS or custom (which I guess would be the default)
- Inputs can be activated for all site level intervals, whether they originate from the project or are custom defined for the site.
   - This is what is currently causing me a headache - say there is a project level interval. What do we do when the user activates the allow input switch? The project intervals aren't saved on the backend / don't have the input information if they are. 
   - My thought is that whenever a user adds a site to a project, we will need to create the site level intervals that correspond to the project preset. This would make sure that the required site level intervals are present.
   - The downside of that is that it seems to require a code we have to write populating and deleting these site level intervals that seems like it should be handled by SQL.
   - Also, it seems to contradict how the `SoilScreen` component is currently written, where a site's list of site intervals is taken as a union of the project level intervals with the site level intervals.

I think that's about it.

**Comment by shrouxm** [comment]
@david-code my initial thought is: what if we move the `$INPUT_enabled` fields from the `SoilDataDepthInterval` model to the `DepthDependentSoilData` model?

i think it resolves the headache you're describing above. but it violates the mental model that the `SoilData` models are for the actual data and the other ones are for configuring the data inputs

can you think of any edge cases that aren't handled if we do that?

we could maintain the mental model by adding a `DepthDependentConfiguration` model just to hold the `$INPUT_enabled` fields, but that is more work and i'm not sure it matters

**Comment by david-code** [comment]
@shrouxm Good idea! I'll think about it a bit and see if any problems come up. If not, I'll give it a shot.

**Comment by david-code** [comment]
Added some notes to the [backend PR](https://github.com/techmatters/terraso-backend/pull/1100#issue-2077640369) on implementation.

**Comment by ltseng** [comment]
Practically done! Garo and David spoke last week and are in agreement for how to move forward. Looking forward to demo tomorrow!

**Comment by shrouxm** [comment]
i will check on this today, i'm not totally sure. there may still be small bugs or UI mismatches before it's ready for QA

**Comment by shrouxm** [comment]
@ltseng FYI status update: i finally looked through this and made a new list of remaining work. i asked courtney some questions in figma and will work on this tomorrow once i'm clear on everything!

**Comment by ltseng** [comment]
Awesome, thanks!

**Comment by ltseng** [comment]
Adding a bit more detail for what I observed in the bug with modifying existing depth intervals - I created two, 0-10 and 15-20. When I modified the first to 0-8 instead, the app crashed, and the next time I opened the site, I was left with the 15-20 depth intervals only. The app had somehow deleted my first one, which I was trying to modify.

(Skip to 0:50)
https://github.com/techmatters/terraso-mobile-client/assets/2730009/b1b0a5e0-bd47-4cbb-91c9-fb022fac01d7

![image](https://github.com/techmatters/terraso-mobile-client/assets/2730009/b0943e55-7da0-45f1-b307-53a7bce7d9de)

**Comment by ltseng** [comment]
Also, I'm noticing that after I've created a custom depth interval already, the modal to create a new depth interval ends up showing the previous values I had typed in when I try to create a new one

![image](https://github.com/techmatters/terraso-mobile-client/assets/2730009/eab1fa2a-49e0-4090-a7b2-b20c4c745c1b)

**Comment by shrouxm** [comment]
hmm looks like the modifying a depth interval crash behaves differently for you, both good catches, thank you!

**Comment by ltseng** [comment]
I notice too that I'm unable to edit depth intervals after they've been created at the project level, even when logged in as a manager. Is this intentional, or not yet implemented?

![image](https://github.com/techmatters/terraso-mobile-client/assets/2730009/d4dfc450-786f-4479-9b17-1b8f0940840d)

**Comment by shrouxm** [comment]
that is intentional, the design doesn't accommodate for changing project level depth intervals. i think because:
- at the site level, you still bring along the settings for which inputs are enabled if not the data, so there is a behavioral difference vs deleting & recreating
- at the site level, the worst thing you could do is delete the data for one depth interval so it's low-risk if someone gets confused. at the project level, you could be accidentally deleting the data collected at every site for that depth interval if you misunderstand what's going on, so we want to be as simple as possible to prevent confusion

as an aside, it would not be hard to convince me that site level depth intervals also should not be editable, but right now the design is for only non-preset site-level depth intervals to be editable

**Comment by ltseng** [comment]
Thanks for the explanation! I agree with this reasoning - just wanted to make sure so that I can revise the test cases accordingly!

**Comment by ltseng** [comment]
Quick note for your reference @shrouxm - when I create a depth interval within an individual site where the project is set to custom/no depth intervals, that depth interval is populated by default with texture/rock fragments/color/structure, unlike when the project is set to NRCS/LandPKS depth intervals.

**Comment by shrouxm** [comment]
moved the remaining work to a new issue #975 so i can close this one
